### PR TITLE
Rename all test assemblies to include a library prefix

### LINKED ISSUE #744: Top level bin\ output folder has become cluttered with overly generic names
Many of the files below are specific to a given library's tests, but there is no convention to identify them as such. e.g. 'Assembly1' is specific to FileVersionInfo tests and 'ModuleCore' is for Xml. We should either:

1. Rename all test assemblies to include a prefix matching their parent.
2. Create a hierarchical structure in bin\ 

I think 1 would be better as it would also make the build log easier to understand.

I also question whether we even need so many test assemblies, especially in the XML case.

``
D:\Src\corefx\bin\Debug>dir /b * | findstr /v "System" | findstr /v "Microsoft"
Assembly1
Axes.Tests
BaseLibManaged
Events.Tests
MemoryMappedFile.Tests
MemoryMappedViewAccessor.Tests
MemoryMappedViewStream.Tests
Misc.Tests
ModuleCore
NativeTestService
ProcessTest_ConsoleApp
Properties.Tests
SDMSample.Tests
Streaming.Tests
TreeManipulation.Tests
XDocument.Common
XDocument.Test.ModuleCore
XmlCoreTest
XmlDiff
xNodeBuilder.Tests
xNodeReader.Tests
XunitTraitsDiscoverers
```

### PR DESCRIPTION
**Pull request by Maxwe11**
Fixes #744

Open question: what to do with 
- BaseLibManaged
- ModuleCore
- XDocument.Common
- XDocument.Test.ModuleCore
- XmlCoreTest
- XmlDiff

### CODE REVIEW

File: src/System.Diagnostics.Process/tests/System.Diagnostics.Process.Tests/System.Diagnostics.Process.Tests.csproj [hunk 1]
```diff
@@ -32,9 +32,9 @@
       <Project>{be8ed8c1-c314-4c4e-a929-64c9c8b3552a}</Project>
       <Name>XunitTraitsDiscoverers</Name>
     </ProjectReference>
-    <ProjectReference Include="..\ProcessTest_ConsoleApp\ProcessTest_ConsoleApp.csproj">
+    <ProjectReference Include="..\ProcessTest_ConsoleApp\System.Diagnostics.Process.TestConsoleApp.csproj">
       <Project>{69e46a6f-9966-45a5-8945-2559fe337827}</Project>
-      <Name>ProcessTest_ConsoleApp</Name>
+      <Name>System.Diagnostics.Process.TestConsoleApp</Name>
```
Comment (krwq) [inline src/System.Diagnostics.Process/tests/System.Diagnostics.Process.Tests/System.Diagnostics.Process.Tests.csproj]: I believe this name might be hard coded inside the test itself. Please clean your workspace and check if the test is not complaining that the exe file is missing.

Comment (Maxwe11) [inline src/System.Diagnostics.Process/tests/System.Diagnostics.Process.Tests/System.Diagnostics.Process.Tests.csproj]: Yes, it's hard coded in code, and I left old assembly name `ProcessTest_ConsoleApp.exe`, just project name changed.

Comment (krwq) [inline src/System.Diagnostics.Process/tests/System.Diagnostics.Process.Tests/System.Diagnostics.Process.Tests.csproj]: I can still see the new assembly name in the sln file.

**Comment by ellismg** [comment]
@krwq, can you take another look?

File: src/System.ServiceProcess.ServiceController/tests/System.ServiceProcess.ServiceController.Tests/System.ServiceProcess.ServiceController.Tests.csproj [hunk 2]
```diff
@@ -29,9 +29,9 @@
     </ProjectReference>
   </ItemGroup>
   <ItemGroup Condition="'$(TargetsWindows)'=='true'">
-    <ProjectReference Include="..\NativeTestService\NativeTestService.vcxproj"> 
+    <ProjectReference Include="..\NativeTestService\System.ServiceProcess.ServiceController.TestNativeService.vcxproj">
```
Comment (weshaggard) [inline src/System.ServiceProcess.ServiceController/tests/System.ServiceProcess.ServiceController.Tests/System.ServiceProcess.ServiceController.Tests.csproj]: This test project is also calling create process with this referenced exe so the tests will break with this rename.

File: src/System.Diagnostics.Process/tests/System.Diagnostics.Process.Tests/System.Diagnostics.Process.Tests.csproj (hunk 1 above)
Comment (weshaggard) [inline src/System.Diagnostics.Process/tests/System.Diagnostics.Process.Tests/System.Diagnostics.Process.Tests.csproj]: I think it is actually OK to rename this to System.Diagnostics.Process.TestConsolApp you just need to update the tests to use this new process name.

**Comment by weshaggard** [comment]
Thanks for updating these @Maxwe11, the changes look good to me.

**Comment by Maxwe11** [comment]
@weshaggard may be do the same for assemblies? And also rename project folder because now project name doesn't match folder name where it placed e.g. project `System.Xml.XDocument.xNodeBuilder.Tests.csproj` lives in `xNodeBuilder` folder.
What your thoughts?

**Comment by weshaggard** [comment]
Yes I do think we should make the directory names consistent as well.

**Comment by stephentoub** [comment]
LGTM2.  Thanks for the cleanup.

**Pull request merged**